Automattic/wp-calypso
# Signup: detect WordPress, Medium, Blogger, and Squarespace sites for import

#### Changes proposed in this Pull Request

* Updates the undocumented `isSiteImportable` call to use the new endpoint from D26042-code.
* Handles the case of a null `site_engine` or empty `importer_types` response by dispatching an error (`SITE_IMPORTER_ERR_BAD_REMOTE`).

#### Testing instructions

* Open the network inspector.
* Start from `/start/import`.
* Enter the url of a url-based importer and continue.
* You should see a request to `/wpcom/v2/imports/is-site-importable` for the url you entered. The response should contain `"importer_types":["url"]`.
* The signup process should continue and you should be taken to Calypso and land directly on the preview screen for starting your import.
* Again, start from `/start/import`.
* Enter the url of a file-based importer and continue.
* You should see a request to `/wpcom/v2/imports/is-site-importable` for the url you entered. The response should contain `"importer_types":["file"]`.
* The signup process should continue and you should be taken to Calypso and land directly on the file upload page for the importer matching the given url. For example, if your url was for a WordPress site, you should be landed on the WordPress importer.
* Again, start from `/start/import`.
* Enter an invalid URL, or one that doesn't have a supported import type, such as example.com.
* When you attempt to continue, you should receive an error "We're not able to reach that site at this time, or it's not compatible with our URL importer."
* You should see a request to `/wpcom/v2/imports/is-site-importable` for the url you entered. The response should contain `"importer_types":[]`, and `site_engine: null`.

Addresses https://github.com/Automattic/samus-private/issues/153
